core/trino-main/src/main/java/io/trino/operator/output/PagePartitionerPool.java

```diff
@@ -74,12 +77,25 @@ public void release(PagePartitioner pagePartitioner)
     public void close()
```

Comment:
It would be cleaner and semantically correct if `PagePartitionerPool` maintined it's own `AggregatedMemoryPool` (see `AggregatedMemoryContext#newAggregatedMemoryContext`) and you close it only after `free.clear();`

`pagePartitionerSupplier` would actually become `Function<AggregatedMemoryContext, PagePartitioner>` then

Reply:
every `PagePartitioner` manages its own memory and has to ensure it's released because `PagePartitioner` can be closed in regular operation.
So after `closeSafely` call below (given the buffers nullification added above) we are guaranteed that all memory is released.
This I think makes keeping `AggregatedMemoryPool` in the `PagePartitionerPool` redundant